frontend/public/components/utils/firehose.jsx

```diff
@@ -25,8 +25,12 @@ const processReduxId = ({k8s}, props) => {
   }
 
   if (!isList) {
-    const stuff = k8s.get(reduxID);
-    return stuff ? stuff.toJS() : {};
+    let stuff = k8s.get(reduxID);
+    if (stuff) {
+      stuff = stuff.toJS();
+      stuff.ignoreIfMissing = props.ignoreIfMissing;
+    }
+    return stuff || {};
```

Comment:
can we use the same approach which is used for lists?

```js
const data = k8s.getIn([reduxID, 'data']);

 return {
    data: data && data.toJS(), 
    loadError: k8s.getIn([reduxID, 'loadError']),	
    loaded: k8s.getIn([reduxID, 'loaded']),
    optional: props.optional,
    ignoreIfMissing: props.ignoreIfMissing,
};
```

This way we can optionally swap the `toJS` with only immutables once they will are needed in the future.

Reply:
Actually the `stuff` is not response received from API but already a wrapper, please see https://github.com/openshift/console/blob/98211bbd03cacfb11176407953902506f19d5e40/frontend/public/reducers/k8s.ts#L135

Reply:
I know, but it will be easier to extend for adding a support of raw immutables

Reply:
meaning that `toJS` may not be used at all

Reply:
If this is needed, I propose to do it in a follow-up.
It would require refactoring of code around the reducer and this is unrelated to original need behind this PR.